File: iosApp/iosApp/Components/BottomTabBar.swift
import SwiftUI

enum AppTab: CaseIterable {
    case home, myPosts, profile

    var labelKey: String {
        switch self {
        case .home: return "tab_home"
        case .myPosts: return "tab_my_posts"
        case .profile: return "tab_profile"
        }
    }

    var icon: String {
        switch self {
        case .home: return "house.fill"
        case .myPosts: return "square.and.pencil"
        case .profile: return "person.fill"
        }
    }
}

struct BottomTabBar: View {
    @Binding var selectedTab: AppTab

    @Environment(\.appColors) private var colors
    @EnvironmentObject var lang: LanguageManager

    var body: some View {
        VStack(spacing: 0) {
            Rectangle()
                .fill(colors.divider)
                .frame(height: 0.5)

            HStack {
                ForEach(AppTab.allCases, id: \.self) { tab in
                    Button {
                        selectedTab = tab
                    } label: {
                        VStack(spacing: 4) {
                            Image(systemName: tab.icon)
                                .font(.system(size: 20))
                            Text(lang.s(tab.labelKey))
                                .font(.system(size: 11))
                        }
                        .frame(maxWidth: .infinity)
                        .foregroundColor(
                            selectedTab == tab
                                ? colors.tabActive
                                : colors.tabInactive
                        )
                    }
                    .buttonStyle(.plain)
                }
            }
            .padding(.top, 8)
            .padding(.bottom, 4)
        }
        .background(colors.surface)
    }
}

#Preview {
    BottomTabBar(selectedTab: .constant(.home))
        .environmentObject(LanguageManager.shared)
}

File: androidApp/src/main/kotlin/com/example/app/android/components/BottomTabBar.kt
package com.example.app.android.components

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.interaction.MutableInteractionSource
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.WindowInsets
import androidx.compose.foundation.layout.asPaddingValues
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.navigationBars
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.remember
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.annotation.StringRes
import androidx.compose.ui.res.stringResource
import com.example.app.android.R
import com.example.app.android.theme.AppTheme

enum class Tab(@StringRes val labelRes: Int, val iconRes: Int) {
    Home(R.string.tab_home, R.drawable.ic_tab_home),
    MyPosts(R.string.tab_my_posts, R.drawable.ic_tab_my_posts),
    Profile(R.string.tab_profile, R.drawable.ic_tab_profile)
}

@Composable
fun BottomTabBar(
    selectedTab: Tab,
    onTabSelected: (Tab) -> Unit,
    modifier: Modifier = Modifier
) {
    val colors = AppTheme.colors
    Column(
        modifier = modifier
            .background(colors.surface)
    ) {
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .height(0.5.dp)
                .background(colors.divider)
        )
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(top = 8.dp, bottom = 4.dp)
                .padding(WindowInsets.navigationBars.asPaddingValues()),
            horizontalArrangement = Arrangement.SpaceEvenly
        ) {
            Tab.entries.forEach { tab ->
                val isSelected = selectedTab == tab
                Column(
                    horizontalAlignment = Alignment.CenterHorizontally,
                    modifier = Modifier
                        .clickable(
                            interactionSource = remember { MutableInteractionSource() },
                            indication = null,
                            onClick = { onTabSelected(tab) }
                        )
                        .weight(1f)
                ) {
                    val label = stringResource(tab.labelRes)
                    Icon(
                        painter = painterResource(id = tab.iconRes),
                        contentDescription = label,
                        modifier = Modifier.size(24.dp),
                        tint = if (isSelected) colors.tabActive else colors.tabInactive
                    )
                    Text(
                        text = label,
                        fontSize = 11.sp,
                        color = if (isSelected) colors.tabActive else colors.tabInactive
                    )
                }
            }
        }
    }
}
